ephios/plugins/basesignup/signup/section_based.py

```diff
@@ -0,0 +1,319 @@
+    def __init__(self, data=None, **kwargs):
+        super().__init__(data, **kwargs)
+        self.sections_formset = SectionsFormset(
+            data=data,
+            initial=self.initial.get("sections", list()),
+            prefix="sections",
+        )
+
+    def clean_sections(self):
+        if not self.sections_formset.is_valid():
+            raise ValidationError(_("The sections aren't configured correctly."))
+
+        sections = [
+            {
+                key: form.cleaned_data[key]
+                for key in ("title", "qualifications", "min_count", "uuid")
+            }
+            for form in self.sections_formset
+        ]
+        return sections
+
+
+class SectionSignupForm(forms.Form):
+    section = forms.ChoiceField(
+        label=_("Preferred Section"),
+        widget=forms.RadioSelect,
+        # choices are set as (uuid, title) of section
+    )
+
+
+class SectionBasedSignupView(FormView, BaseSignupView):
+    template_name = "basesignup/section_based/signup.html"
+
+    def get_form(self, form_class=None):
+        form = SectionSignupForm(self.request.POST)
+        form.fields["section"].choices = [
+            (section["uuid"], section["title"])
+            for section in sections_participant_qualifies_for(
+                self.method.configuration.sections, self.request.user.as_participant()
+            )
+        ]
+        return form
+
+    def get_context_data(self, **kwargs):
+        kwargs.setdefault("shift", self.shift)
+        kwargs.setdefault(
+            "unqualified_sections",
+            [
+                section["title"]
+                for section in self.method.configuration.sections
+                if section
+                not in sections_participant_qualifies_for(
+                    self.method.configuration.sections, self.request.user.as_participant()
+                )
+            ],
+        )
+        return super().get_context_data(**kwargs)
+
+    def form_valid(self, form):
+        try:
+            with transaction.atomic():
+                self.method.perform_signup(
+                    self.request.user.as_participant(),
+                    preferred_section_uuid=form.cleaned_data["section"],
+                )
+                messages.success(
+                    self.request,
+                    self.method.signup_success_message.format(shift=self.shift),
+                )
+        except ParticipationError as errors:
+            for error in errors:
+                messages.error(self.request, self.method.signup_error_message.format(error=error))
+        return redirect(self.shift.event.get_absolute_url())
+
+    def signup_pressed(self, request, *args, **kwargs):
+        if not self.method.configuration.choose_preferred_section:
+            return super().signup_pressed(request, *args, **kwargs)
+        return redirect(reverse("event_management:shift_action", kwargs=dict(pk=self.shift.pk)))
```

Comment:
was just confused because of the "shift_action" URL. maybe we should rename this to smth like signup_action?